### PR TITLE
[MD]Severless Support: update `data-test-subj`attribute for functional tests & fix bug in edit flow

### PR DESCRIPTION
**Pull request by zhongnansu**
### Description

[MD]Update data-test-subj for functional tests & fix bug in edit flow

related PR in functional test repo: **https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/669**
### Issues Resolved

https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/665
### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by zhongnansu** [comment]
Skipping changelog, because the feature is not released yet

**Review by kristenTian — APPROVED** [review]

**Review by kristenTian — APPROVED** [review]

**Review by kristenTian — APPROVED** [review]

**Review by kristenTian — APPROVED** [review]

**Review by kristenTian — APPROVED** [review]

**Review by ZilongX — APPROVED** [review]

File: src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx [hunk 1]
```diff
@@ -430,7 +430,7 @@ export class CreateDataSourceForm extends React.Component<
                 value={this.state.auth.credentials.service}
                 onChange={(e) => this.onChangeSigV4ServiceName(e)}
                 name="ServiceName"
-                data-test-subj="createDataSourceFormAuthTypeSelect"
+                data-test-subj="createDataSourceFormSigV4ServiceTypeSelect"
```
Comment (Flyingliuhub) [inline src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx:433]: Do you need to modify file create_data_source_form.test.tsx? Thanks

File: src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx [hunk 2]
```diff
@@ -804,7 +819,7 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
             value={this.state.auth.credentials?.region || ''}
             onChange={this.onChangeRegion}
             onBlur={this.validateRegion}
-            data-test-subj="updateDataSourceFormRegionField"
+            data-test-subj="editDataSourceFormRegionField"
```
Comment (Flyingliuhub) [inline src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx:822]: Do you need to modify the file "edit_data_source_form.test.tsx" ? Thanks

Comment (zhongnansu) [reply src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx:822]: "data-test-subj" attributes are for cypress tests only. I have a PR open in the cypress test repo using those updated values. https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/665

**Review by Flyingliuhub — APPROVED** [review]

**Pull request merged**